google_symptoms/delphi_google_symptoms/run.py

```diff
@@ -46,62 +40,21 @@ def run_module(params):
     oldest_final_export_date = None
-
-    export_start_date = datetime.strptime(
-        params["indicator"]["export_start_date"], "%Y-%m-%d")
-    # If end_date not specified, use current date.
-    export_end_date = datetime.strptime(
-        params["indicator"].get(
-            "export_end_date", datetime.strftime(date.today(), "%Y-%m-%d")
-        ), "%Y-%m-%d")
-
     export_dir = params["common"]["export_dir"]
-    num_export_days = params["indicator"]["num_export_days"]
-
-    if num_export_days is None:
-        # Generate a list of signals we expect to produce
-        sensor_names = set(
-            "_".join([metric, smoother, "search"])
-            for metric, smoother in product(COMBINED_METRIC, SMOOTHERS)
+    # safety check for when patch parameter is still in the json, but not running patch
+    patch_flag = False if not params.get("patch") else params["patch"].get("patch_flag", False)
+
+    if logger is None:
+        logger = get_structured_logger(
+            __name__,
+            filename=params["common"].get("log_filename"),
+            log_exceptions=params["common"].get("log_exceptions", True),
         )
 
-        # Fetch metadata to check how recent each signal is
-        covidcast.use_api_key(params["indicator"]["api_credentials"])
-        metadata = covidcast.metadata()
-        # Filter to only those we currently want to produce, ignore any old or deprecated signals
-        gs_metadata = metadata[(metadata.data_source == "google-symptoms") &
-            (metadata.signal.isin(sensor_names))]
-
-        if sensor_names.difference(set(gs_metadata.signal)):
-            # If any signal not in metadata yet, we need to backfill its full history.
-            num_export_days = "all"
-        else:
-            # Calculate number of days based on what's missing from the API and
-            # what the validator expects.
-            max_expected_lag = lag_converter(
-                params["validation"]["common"].get("max_expected_lag", {"all": 4})
-                )
-            global_max_expected_lag = max( list(max_expected_lag.values()) )
-
-            # Select the larger number of days. Prevents validator from complaining about
-            # missing dates, and backfills in case of an outage.
-            num_export_days = max(
-                (datetime.today() - to_datetime(min(gs_metadata.max_time))).days + 1,
-                params["validation"]["common"].get("span_length", 14) + global_max_expected_lag
-                )
-    if num_export_days == "all":
-        num_export_days = (export_end_date - export_start_date).days + 1
-
-    logger = get_structured_logger(
-        __name__, filename=params["common"].get("log_filename"),
-        log_exceptions=params["common"].get("log_exceptions", True))
-
+    start_date, end_date, num_export_days = generate_export_dates(params, logger)
+    export_date_range = generate_query_dates(start_date, end_date, num_export_days, patch_flag)
```

Comment:
issue: Also not sure this is safe to run on patch dates twice -- we're already calling `generate_query_dates` inside `generate_patch_dates` (`date_utils.py:60`). Maybe take the `generate_query_dates` call out of `generate_patch_dates`?

This could also be put into the `if not patch_flag` block, but then we won't get dates in the date range format.

Perhaps put it into the `if not patch_flag` block and also do
```suggestion
    start_date, end_date = generate_query_dates(start_date, end_date, num_export_days, patch_flag)
```

`export_date_range` is only used in one place, the call to `pull_gs_data`, so it's not too important to keep around.

On the other hand, the suggestion above does give new values to `start_date` and `end_date` in the non-patch case, which is probably undesirable (not sure how the logic works out compared to the old version of the code, but this particular date processing used to happen inside of `pull_gs_data` and therefore didn't update the top-level start and end date values).

Reply:
I think there are lots of ways to address this, but at its heart the issue is that the date processing when doing a patch and _not_ doing a patch aren't parallel. What we get out at the end has different amounts of padding added/subtracted and is in a different format.

Reply:
hmmm I see your point, but at the same time I feel like there could be times where someone wants to do a custom run that's not running the patch specifically. In that case we would still want to run generate_query_dates. I'm going to change some of the variables to better reflect this. 

After the change, if you still think I should change it back, I can.